s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/DoiLinkUpdateEventEmitter.java

```diff
@@ -0,0 +1,84 @@
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.model.Publication;
+import no.unit.nva.publication.service.impl.ResourceService;
+import no.unit.nva.s3.S3Driver;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.StringUtils;
+import nva.commons.core.ioutils.IoUtils;
+import nva.commons.core.paths.UriWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.s3.S3Client;
+
+public class DoiLinkUpdateEventEmitter implements RequestStreamHandler {
+
+    public static final String NEW_LINE_REGEX = "\n";
+    private static final Logger logger = LoggerFactory.getLogger(DoiLinkUpdateEventEmitter.class);
+    private final S3Client s3Client;
+    private final ResourceService resourceService;
+
+    @JacocoGenerated
+    public DoiLinkUpdateEventEmitter() {
+        this(defaultS3Client(), ResourceService.defaultService());
+    }
+
+    public DoiLinkUpdateEventEmitter(S3Client s3Client, ResourceService resourceService) {
+        this.s3Client = s3Client;
+        this.resourceService = resourceService;
+    }
+
+    @Override
+    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
+        attempt(() -> parseInput(input))
+            .map(this::getFileContainingIdentifiers)
+            .map(this::extractIdentifiers).get()
+            .forEach(this::moveDoiToLinkFieldForPublicationWithCorrespondingIdentifier);
+    }
+
+//    private static Publication constructNewPublicationWithUpdatedReferenceDoi(Publication publication) {
+//
+//    }
+
+    private void moveDoiToLinkFieldForPublicationWithCorrespondingIdentifier(String identifier) {
+        attempt(() -> identifier.replaceAll(" ", StringUtils.EMPTY_STRING))
+            .map(SortableIdentifier::new)
+            .map(resourceService::getPublicationByIdentifier)
+            .map(this::moveDoiToLinkField)
+            .map(resourceService::updatePublication);
+    }
+
+    private Publication moveDoiToLinkField(Publication publication) {
+        publication.getEntityDescription().getReference().setDoi(publication.getDoi());
+        publication.setDoi(null);
+        return publication;    }
+
+    private List<String> extractIdentifiers(String string) {
+        logger.info(string);
+        return Arrays.asList(string.split(NEW_LINE_REGEX));
+    }
+
+    private String getFileContainingIdentifiers(EventReference eventReference) {
+        S3Driver s3Driver = new S3Driver(s3Client, eventReference.extractBucketName());
+        var fileLocation = UriWrapper.fromUri(eventReference.getUri()).toS3bucketPath();
+        logger.info(fileLocation.toString());
+        return s3Driver.getFile(fileLocation);
+    }
+
+    private EventReference parseInput(InputStream input) {
+        String inputString = IoUtils.streamToString(input);
+        return attempt(() -> EventReference.fromJson(inputString))
+                   .orElseThrow();
+    }
+}
```

Comment:
A few things to note here, 
- I have moved the code around a bit to make it clearer what the process is, trying to group related operations
- I got rid of several uses of regular expressions (always a code smell ☠️) using File::lines and String::trim (the latter being what I suspect you wanted)
- I removed commented code (🌧️)
- I formatted the code (🌧️)
- Take a look and see if you agree if it is more readable :) 

```
public class DoiLinkUpdateEventEmitter implements RequestStreamHandler {

    private static final Logger logger = LoggerFactory.getLogger(DoiLinkUpdateEventEmitter.class);
    private final S3Client s3Client;
    private final ResourceService resourceService;

    @JacocoGenerated
    public DoiLinkUpdateEventEmitter() {
        this(defaultS3Client(), ResourceService.defaultService());
    }

    public DoiLinkUpdateEventEmitter(S3Client s3Client, ResourceService resourceService) {
        this.s3Client = s3Client;
        this.resourceService = resourceService;
    }

    @Override
    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
        fetchIdentifiersFromInputEvent(input)
            .map(this::fetchPublication)
            .forEach(this::updatePublication);
    }

    private Publication fetchPublication(SortableIdentifier identifier) {
        return attempt(() -> resourceService.getPublicationByIdentifier(identifier)).orElseThrow();
    }
    
    private void updatePublication(Publication publication) {
        var update = moveDoiToLinkField(publication);
        resourceService.updatePublication(update);
    }

    private Publication moveDoiToLinkField(Publication publication) {
        publication.getEntityDescription().getReference().setDoi(publication.getDoi());
        publication.setDoi(null);
        return publication;
    }

    private Stream<SortableIdentifier> fetchIdentifiersFromInputEvent(InputStream inputStream) {
        var eventReference = parseInput(inputStream);
        var s3Driver = new S3Driver(s3Client, eventReference.extractBucketName());
        var fileLocation = UriWrapper.fromUri(eventReference.getUri()).toS3bucketPath();
        logger.info(fileLocation.toString());
        return s3Driver.getFile(fileLocation)
                   .lines()
                   .map(String::trim)
                   .map(SortableIdentifier::new);
    }

    private EventReference parseInput(InputStream input) {
        var inputString = IoUtils.streamToString(input);
        return attempt(() -> EventReference.fromJson(inputString)).orElseThrow();
    }
}
```

Reply:
Agree that this code is more readable :)